### PR TITLE
LLM: Update IPEX to 2.2.0+cpu and Refactor for _ipex_optimize

### PR DESCRIPTION
**Pull request by xiangyuT**
## Description

Update IPEX for `BIGDL_IPEX_OPT` to `2.2.0+cpu`.
Refactor `_ipex_optimize`.

### CODE REVIEW

**Comment by glorysdj** [comment]
also need to update examples  Accelerate with BIGDL_OPT_IPEX part

**Comment by glorysdj** [comment]
need to consider ipex optimization for both int4 & bf16

File: python/llm/example/CPU/Speculative-Decoding/llama2/README.md [hunk 1]
```diff
@@ -98,27 +98,13 @@ First token latency xx.xxxxs
...
-```
-
-#### 4.4 Install IPEX and other dependencies
-```bash
-# Install IPEX 2.3.0+git0c63936
-bash compile_bundle.sh 
+# Install IPEX 2.2.0+cpu
+python -m pip install torch==2.2.0 torchvision==0.17.0 torchaudio==2.2.0 --index-url https://download.pytorch.org/whl/cpu
+python -m pip install intel-extension-for-pytorch==2.2.0
+python -m pip install oneccl_bind_pt==2.2.0 --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/cpu/us/
```
Comment (Uxito-Ada) [inline python/llm/example/CPU/Speculative-Decoding/llama2/README.md:107]: I suggest to install oneccl binding with the wheel in https://developer.intel.com/ipex-whl-stable-cpu, as I could not access the US link even on our machine

**Review by glorysdj — APPROVED** [review]
LGTM

**Comment by Uxito-Ada** [comment]
Tried to reproduce accept rate of Baichuan2-7B dropping to zero issue, but no problem happened anymore:

```bash
<bot>霍·阿·布恩蒂亚是一个充满激情和冒险精神的男子，他对新事物充满好奇，对新世界充满期待。当他第一次见到吉卜赛人带来的磁铁时，他被这个神奇的发明深深吸引，决定用自己的一匹骡子和两只山羊换取两块磁铁。他认为磁铁可以帮助他从这片崭新的天地中找到金子，从而振兴破败的家业。

吉卜赛人带来了许多神奇的东西，其中最引人注目的是磁铁。他们向村民们展示了磁铁的神奇力量，让铁锅、铁盆、铁钳、铁炉等都从原地倒下，木板上的钉子和螺丝嘎吱嘎
Tokens generated 128
......
Draft len: 5.24, accept len: 4.08
```

**Environment:**
spr-14 python 3.9
transformers==4.35.2
torch==2.2.0
intel_extension_for_pytorch==2.2.0
bigdl-llm==2.5.0b20240221 (before inserting PR codes) and inserted PR codes

I think the previous issue is possibly because Baichuan2-7B model had not been modified appropriately.

**Review by Uxito-Ada — APPROVED** [review]
lgtm

**Pull request merged**